care/facility/api/viewsets/patient.py

```diff
@@ -130,11 +136,38 @@ def icmr_sample(self, *args, **kwargs):
         patient.__class__ = PatientIcmr
         return Response(data=PatientICMRSerializer(patient).data)
 
+    @action(detail=True, methods=["POST"])
+    def discharge_patient(self, request, *args, **kwargs):
+        discharged = bool(request.data.get("discharge", False))
+        patient = self.get_object()
+        patient.is_active = discharged
+        patient.allow_transfer = not discharged
+        patient.save()
+        return Response(status=status.HTTP_200_OK)
+
+    @action(detail=True, methods=["POST"])
+    def discharge_summary(self, request, *args, **kwargs):
+        email = request.data.get("email", "")
+        try:
+            validate_email(email)
+        except:
+            return Response({"email": "Invalid Email Provided"}, status=status.HTTP_400_BAD_REQUEST)
+        patient = self.get_object()
+        generate_discharge_report.delay(patient.id, email)
+        return Response(status=status.HTTP_200_OK)
+
     @action(detail=True, methods=["POST"])
     def transfer(self, request, *args, **kwargs):
         patient = PatientRegistration.objects.get(
             id=PatientSearch.objects.get(external_id=kwargs["external_id"]).patient_id
         )
+
+        if patient.allow_transfer == False:
```

Comment:
Codacy found an issue: [Comparison to False should be 'not expr'](https://app.codacy.com/gh/coronasafe/care/pullRequest?prid=5765271)